dexter/src/main/java/com/karumi/dexter/listener/PermissionListener.java

```diff
@@ -32,6 +32,16 @@
   void onPermissionGranted(String permission);
 
   /**
+   * Method called the first time a requested permission has been denied.
+   *
+   * @param permission The permission that has been requested. One of the values found in {@link
+   * android.Manifest.permission}
+   * @param token Token used to continue or cancel the permission request process. The permission
+   * request process will remain blocked until one of the token methods is called.
+   */
+  void onFirstTimePermissionDenied(String permission, PermissionToken token);
```

Comment:
I'd keep the listener as simple as possible, adding methods forces subclasses to implement something clients might not need at all. Besides, it looks somewhat confusing for people that is using the library for the first time, if the user denies the permission for the first time, should I expect to be informed with ``onPermissionDenied`` as well or only in one of the methods?

There is a cleaner solution that might be useful in the future for implementing other features: Provide listeners a ``RequestedPermission`` object with additional information instead of a simple String. This object might contain the name of the permission, the number of times that the permission has been requested/denied (using a boolean for tracking the first time seems pretty arbitrary to me), if the permission has been *perma-denied*, etc.
With this approach we can add more features while keeping backwards compatibility (once we start sending the new object ofc).